lib/auth/methods.go

```diff
@@ -378,6 +385,22 @@ func (a *Server) authenticateUserInternal(
 		return a.authenticatePasswordless(ctx, req)
 	}
 
+	// Disallow non-local users from local authentication.
+	// Passwordless does its own check, as the user is only known after the
+	// webauthn assertion is cleared.
+	switch u, err := a.GetUser(ctx, user, false /* withSecrets */); {
+	case trace.IsNotFound(err):
+		// Keep going if the user is not known.
```

Comment:
This was the case before and is covered by other tests, so seemed reasonable to keep the behavior.